bot/actions/slack_incident_actions.rb

```diff
@@ -24,8 +24,59 @@ def open_incident(slack_action)
     threads.each(&:join)
   end
 
+  def update_incident(slack_action, channel_id)
+    incident = SlackUpdateModal.new(slack_action)
+    current_summary = current_summary(channel_id)
+
+    updated_summary = incident.summary_text(current_summary)
+
+    current_members = current_members(channel_id)
+
+    user_invite_list = create_member_invite_list(
+      current_members,
+      incident.new_incident_comms_lead,
+      incident.new_incident_tech_lead,
+      incident.new_incident_support_lead,
+    )
+
+    threads = []
+    threads << Thread.new { set_channel_details!(channel_id, updated_summary) }
+
+    unless user_invite_list.compact.empty?
+      threads << Thread.new { invite_users!(channel_id, user_invite_list) }
+    end
+
+    unless updated_summary == current_summary
+      threads << Thread.new { notify_channel_of_update!(channel_id) }
+    end
+
+    threads.each(&:join)
+  end
+
 private
 
+  def create_member_invite_list(current_members, comms_lead, tech_lead, support_lead)
```

Comment:
Could `SlackUpdateModal` do this work, given `current_members`, eg `incident.create_invite_list(current_members)`?